site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,177 @@
+
+{% include_codefile 'demo/finished/biz-e-corp/src/ApplicationContext.ts' lines:32-41 %}
+
+The `formInput` method is providing the same functionality that the `_onFormInput` method in the `App` class and the `submitForm` method is doing the same thing that the `App` class's `_addWorker` is. The implementations are slightly different since the `ApplicationContext` has dedicated fields that are storing the state information. Also, since the `ApplicationContext` is not a widget, it needs to emit an `invalidate` event to inform the application to schedule a render cycle.
+
+Notice that the `ApplicationContext` does not have any code to load state information. Its role is only to manage the application's state. Any data that the application should start with must be provided to the `ApplicationContext` via its `constructor` while the application is being started.
+
+Now that we have moved state management to a dedicated module, we need a way to initialize the state and make it available. We will do that in the next section where we will create and register an `Injector`.
+
+{% section %}
+
+## Injectors
+
+{% task 'Register an `Injector` that will allow state to be injected into widgets.' %}
+
+Currently, the application's `main` module is only responsible for creating the `Projector`, which provides the bridge between the application code and the DOM.
+
+{% include_codefile 'demo/initial/biz-e-corp/src/main.ts' %}
+
+Now, we need to update `main` to construct and populate the `ApplicationContext` as well as register that context so that it is accessible to the rest of the application.
+
+{% instruction 'Import the `ApplicationContext` module and add this code to `main`:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:10-29 %}
+
+{% aside 'Loading data' %}
+In a real-world application, this data would probably be loaded via a call to a web-service or a local datastore. To learn more, take a look at the [data stores tutorial](../comingsoon.html).
+{% endaside %}
+
+This is the same data that we used in the `App` module to initialize the `WorkerProperties`, but it is now in a more appropriate location. In general, the `main` module of an application should be concerned with initializing application-wide state. Also, as we mentioned earlier, the `App` class only needed the `WorkerProperties` because its children needed them.
+
+Now that we have the `ApplicationContext`, it is time to allow it to be used by widgets. To do that, we need to create an `Injector` and add it to the application's central widget registry. `Injectors` are created using the `Injector` factory function from `@dojo/widget-core`.
+
+{% instruction 'Add the following imports to the `main` module.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:2-3 %}
+
+{% instruction 'Now, add the following after the `applicationContext` declaration.' %}
```

Comment:
I'm not sure what you mean here. The `applicationContext` is specifically mentioned. Why do you think its unclear?